src/Compilers/Core/Portable/RuleSet/RuleSet.cs

```diff
@@ -332,7 +332,7 @@ private static ReportDiagnostic GetDiagnosticOptionsFromRulesetFile(Dictionary<s
             }
             catch (IOException e)
             {
-                if (e is FileNotFoundException || e is DirectoryNotFoundException)
+                if (e is FileNotFoundException || e.GetType().Name == "DirectoryNotFoundException")
```

Comment:
@davkean this is intended to be changed to a type check.  The use of a name comparison here is because it follows the pattern laid out in other parts of our code base.  I didn't want to change the pattern in the middle of this move.  It's one of my very next check-ins.